packages/angular_devkit/build_angular/src/angular-cli-files/utilities/bundle-calculator.ts

```diff
@@ -5,22 +5,94 @@
  * Use of this source code is governed by an MIT-style license that can be
  * found in the LICENSE file at https://angular.io/license
  */
+import * as webpack from 'webpack';
 import { Budget } from '../../browser/schema';
 
-export interface Compilation {
-  assets: { [name: string]: { size: () => number } };
-  chunks: { name: string, files: string[], isOnlyInitial: () => boolean }[];
-  warnings: string[];
-  errors: string[];
-}
-
 export interface Size {
   size: number;
   label?: string;
 }
 
-export function calculateSizes(budget: Budget, compilation: Compilation): Size[] {
-  const calculatorMap = {
+export interface Threshold {
+  limit: number;
+  type: ThresholdType;
+  severity: ThresholdSeverity;
+}
+
+export enum ThresholdType {
+  MAX = 'maximum',
+  MIN = 'minimum',
+}
+
+export enum ThresholdSeverity {
+  WARNING = 'warning',
+  ERROR = 'error',
+}
+
+export function* calculateThresholds(budget: Budget): IterableIterator<Threshold> {
```

Comment:
Nice use of a generator.

Reply:
Thanks! Actually just threw in another one to check the budgets in order to share code a little better with the existing Webpack plugin.